### PR TITLE
Parameters check for ResourceIdentifier

### PR DESCRIPTION
**Pull request by JonathanCrd**
Checks for null and out of range value added.
Missing test cases for ResourceIdentifier added as well.

### CODE REVIEW

File: Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs [hunk 1]
```diff
@@ -119,5 +154,53 @@ public void CheckHashCode(bool expected, string resourceId1, string resourceId2)
...
             Assert.AreEqual(expected, resourceIdentifier1.GetHashCode() == resourceIdentifier2.GetHashCode());
         }
+
+        [TestCase(TrackedResourceId, TrackedResourceId, true)]
+        [TestCase(ChildResourceId, ChildResourceId, true)]
+        [TestCase(null, null, true)]
+        [TestCase(TrackedResourceId, ChildResourceId, false)]
+        [TestCase(ChildResourceId, TrackedResourceId, false)]
+        [TestCase(null, TrackedResourceId, false)]
+        public void equals(string resourceProviderID1, string resourceProviderID2, bool expected)
```
Comment (m-nash) [inline Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs:164]: Method should start with capital

File: Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs [hunk 2]
```diff
@@ -119,5 +154,53 @@ public void CheckHashCode(bool expected, string resourceId1, string resourceId2)
...
+            Assert.AreEqual(expected, ResourceIdentifier.Equals(a,b));
+        }
+
+        [TestCase(TrackedResourceId, TrackedResourceId, 0)]
+        [TestCase(TrackedResourceId, ChildResourceId, -1)]
+        [TestCase(ChildResourceId, TrackedResourceId, 1)]
+        [TestCase(TrackedResourceId, null, 1)]
+        [TestCase(null, TrackedResourceId, -1)]
+        [TestCase(null, null, 0)]
+        public void compareToResourceProvider(string resourceProviderID1, string resourceProviderID2, int expected)
```
Comment (m-nash) [inline Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs:180]: Method should start with capital

File: Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs [hunk 3]
```diff
@@ -119,5 +154,53 @@ public void CheckHashCode(bool expected, string resourceId1, string resourceId2)
...
+            Assert.AreEqual(expected, ResourceIdentifier.CompareTo(a, b));
+        }
+
+        [TestCase(TrackedResourceId, TrackedResourceId, 0)]
+        [TestCase(TrackedResourceId, ChildResourceId, -1)]
+        [TestCase(ChildResourceId, TrackedResourceId, 1)]
+        [TestCase(TrackedResourceId, null, 1)]
+        [TestCase(null, TrackedResourceId, -1)]
+        [TestCase(null, null, 0)]
+        public void compareToString(string resourceProviderID1, string resourceProviderID2, int expected)
```
Comment (m-nash) [inline Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs:196]: Method should start with capital

**Review by m-nash — CHANGES_REQUESTED** [review]

File: Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs [hunk 4]
```diff
@@ -18,6 +18,41 @@ public void Setup()
...
+                Assert.IsNull(z);
+                Assert.IsNull(y);
+            }
+            else
+            {
+                Assert.AreEqual(resourceProviderID, y);
+            }
+            
+        }
+
```
Comment (m-nash) [inline Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs:55]: add a new test case which validates passing in null to the public constructor

**Review by m-nash — CHANGES_REQUESTED** [review]

File: Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs (hunk 4 above)
Comment (m-nash) [reply Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs:78]: Doesn't look like this got added

**Review by m-nash — CHANGES_REQUESTED** [review]

File: Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs (hunk 4 above)
Comment (JonathanCrd) [reply Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs:78]: It was added in the last commit

**Review by m-nash — APPROVED** [review]
